students/DennisLee/lesson07/html_render.py

```diff
@@ -0,0 +1,327 @@
+	"""
+	def render(self, file_out, cur_ind=""):
+		"""
+		Create (render) a one-line tag object onto an HTML text object.
+
+		:file_out:  A StringIO or FileIO object representing the output
+		            stream to save the rendered HTML text to.
+
+		:cur_ind:  A string of spaces specifying how much to indent
+		           for this particular `OneLineTag` object.  The default
+				   is to not indent at all.
+
+		:return:  `True` if the method succeeds; otherwise, `None`.
+		"""
+		# Make sure arguments are passed correctly
+		self.verify_render_args(file_out, cur_ind)
+
+		# If `indent` member not currently defined, set it to the passed
+		# indentation argument, and change the current indent level to 0
+		if not self.indent:
+			self.indent, cur_ind = cur_ind, ""
+		for i in range(len(self.contents)):
+			item = self.contents[i]
+			if not isinstance(item, str):
+				raise TypeError(
+						f"Element content item #{i} '{item}' is a "
+						f"{type(item)}; it must be a string "
+						f"since {self.tag} is a one-line element.")
+		file_out.write(cur_ind + '<{0}{1}>{2}</{0}>\n'.format(
+				self.tag, self.attribute_string, ' '.join(self.contents)))
+		return True
+		
+class SelfClosingTag(Element):
+	"""
+	The base class for HTML elements that contain neither child text nor
+	child elements.  These elements do not need a separate closing tag,
+	as you can just add a slash ("/") in the initial tag to indicate
+	that it is self-contained.
+	"""
+	def append(self, content):
+		"""
+		Override the `append` method to make sure nothing can be added
+		as child text or child elements.
+
+		:content:  The child content (text, `Element` objects, or a list
+		           or tuple containing text and/or `Element` objects)
+				   that is proposed for addition.
+
+		:return:  `None` if nothing was specified for addition;
+		          otherwise, a `TypeError` exception is raised.
+		"""
+		if content is not None:
+			raise TypeError(f"Cannot add content '{content}' "
+					f"of type '{type(content)}' to element '{self.tag}', "
+					"because the element is a self-closing tag.")
+	def render(self, file_out, cur_ind=""):
+		"""
+		Create (render) a self-closing tag object onto an HTML text
+		object.
+
+		:file_out:  A StringIO or FileIO object representing the output
+		            stream to save the rendered HTML text to.
+
+		:cur_ind:  A string of spaces specifying how much to indent
+		           for this particular `SelfClosingTag` object.  The
+				   default is to not indent at all.
+
+		:return:  `True` if the method succeeds; otherwise, `None`.
+		"""
+		# Make sure arguments are passed correctly
+		self.verify_render_args(file_out, cur_ind)
+
+		# If `indent` member not currently defined, set it to the passed
+		# indentation argument, and change the current indent level to 0
+		if not self.indent:
+			self.indent, cur_ind = cur_ind, ""
+		file_out.write(cur_ind + '<{0}{1} />\n'.format(
+				self.tag, self.attribute_string))
+		return True
```

Comment:
your `render` method shouldn't return a value. I assume this was done for tests purposes, I would generally avoid adding returns just for unit tests. 
You can test your render function by using StringIO class for `file_out` instance and then getting the string written for assert.